Amplify/Categories/DataStore/Query/ModelKey.swift

```diff
@@ -32,24 +32,35 @@ import Foundation
 ///     post.content != nil
 /// })
 /// ```
-public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {}
+public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {
```

Comment:
discussed offline about returning the model name (and use the `ModelRegistry` to resolve the actual model type and its schema) and provide a default implementation to avoid breaking existing models and decouple this PR from the needed CLI changes